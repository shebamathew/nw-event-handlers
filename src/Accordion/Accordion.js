import React, {Fragment} from "react";

class Accordion extends React.Component{
    static defaultProps = {
        sections: [{title:"", content: ""}]
    }; 

    state = {
        currentSectionIndex:0
    };

    handleButtonClick(index) {
        this.setState({currentSectionIndex: index})
    }

    renderTitles() {
        return this.props.sections.map((section, index) => (
            <li key={index}>
                <button onClick={() => this.handleButtonClick(index)}>
                    {section.title}
                </button>
                {this.state.currentSectionIndex === index && this.renderContent(section.content)}
            </li>
        )); 
    }
            
    renderContent(content) {
        return <p>{content}</p>; 
    }

    render() {
        return <ul>{this.renderTitles()}</ul>; 
    }
}

export default Accordion