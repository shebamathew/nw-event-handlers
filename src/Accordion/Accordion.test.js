import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'; 
import Accordion from './Accordion';

describe(`Accordion Component`, () => {
  const sections = [{name:'', content:''}];
  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordion />, div);
    ReactDOM.unmountComponentAtNode(div);
  })
  it('renders the first section by default', () => {
    const tree = renderer.create(<Accordion />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})