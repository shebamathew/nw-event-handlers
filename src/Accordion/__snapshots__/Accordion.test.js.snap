// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component renders the first section by default 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
      
    </button>
    <p>
      
    </p>
  </li>
</ul>
`;
